params.go

```diff
@@ -0,0 +1,135 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"regexp"
+	"strings"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+// Params holds the parameter mappings specified  by user
+type Params map[PName]PValue
+type PName string
+type PValue string
+
+type ParamFile struct {
+	ParamList []string `yaml:"params"`
+}
+
+var pNamePattern = `([A-Za-z_][0-9A-Za-z_]*)`
+
+var pNameRegex = regexp.MustCompile(`^` + pNamePattern + `$`)
+var paramDecRegex = regexp.MustCompile(`{{ *` + pNamePattern + ` *}}`)
+var paramRegex = regexp.MustCompile(pNamePattern + ` *= *(.+?)`)
```

Comment:
not sure why you allow spaces here? it's going to be very weird if someone passes -param "abc   =  23".This would go away if you parse a map from yaml, as it will be very clear whats the key and what the value.